docs/ci.md

```diff
@@ -0,0 +1,68 @@
+# CI for InstructLab SDG
+
+Before running any testing locally, ensure you have run `pip install -r requirements-dev.txt` in your environment.
+
+## Unit tests
+
+Unit tests are designed to test specific SDG components or features in isolation. Generally, new code should be adding or modifying unit tests.
+
+All unit tests currently live in the `tests/unit` directory and are run with [pytest](https://docs.pytest.org/) via [tox](https://tox.wiki/).
+
+To run the unit tests, you can run `tox -e unit` or `tox -e unitcov` if you want to generate coverage metrics as well.
+
+In CI, the tests are run with Python 3.10 - 3.12 on Ubuntu and MacOS runners - you can see the details [here](https://github.com/instructlab/sdg/blob/main/.github/workflows/test.yml)
+
+## Functional tests
+
+Functional tests are designed to test SDG components or features in tandem, but not necessarily as part of a complex workflow. New code may or may not need a functional test but should strive to implement one if possible.
+
+All functional tests currently live in the `tests/functional` directory and are run with [pytest](https://docs.pytest.org/) via [tox](https://tox.wiki/).
+
+We have two types of functional tests - non-GPU and GPU. To run the non-GPU functional tests, you can run `tox -e functional`. To run the GPU functional tests,
+you can run `tox -e py3-functional-gpu`.
+
+In CI, the non-GPU tests are run with Python 3.10 - 3.12 on Ubuntu and MacOS runners - you can see the details [here](https://github.com/instructlab/sdg/blob/main/.github/workflows/test.yml)
+
+The GPU tests are run with Python 3.11 on CentOS runners with access to a single NVIDIA Tesla T4 GPU with 16GB of vRAM - you can see the details [here](https://github.com/instructlab/sdg/blob/main/.github/workflows/functional-gpu-nvidia-t4-x1.yml)
+
+## End-to-end (E2E) tests
+
+InstructLab SDG has several end-to-end jobs that run to ensure compatibility with the [InstructLab Core](https://github.com/instructlab/instructlab) project.
+
+There is currently a ["small" t-shirt size E2E job](https://github.com/instructlab/sdg/blob/main/.github/workflows/e2e-nvidia-t4-x1.yml) and a
+["medium" t-shirt size E2E job](https://github.com/instructlab/sdg/blob/main/.github/workflows/e2e-nvidia-l4-x1.yml).
```

Comment:
Note sure we need to mention t-shirt size in here

Reply:
I decided to remove this section in favor of just pointing people to the table beneath it